DER/DerivativesContracts/RightsAndWarrants.rdf

```diff
@@ -166,6 +310,22 @@
 		<fibo-fnd-utl-av:explanatoryNote xml:lang="en">In contrast to traditional equity warrants, with covered warrants, no new issuance of common stock occurs if the warrant is exercised. The underlying shares of common stock are usually either owned by the issuer of the covered warrants or the issuer has a mechanism, such as owning equity warrants for the underlying shares, through which they can obtain the shares.</fibo-fnd-utl-av:explanatoryNote>
 	</owl:Class>
 	
+	<owl:Class rdf:about="&fibo-der-drc-raw;CurrencyWarrant">
+		<rdfs:subClassOf rdf:resource="&fibo-der-drc-cur;CurrencyDerivative"/>
+		<rdfs:subClassOf rdf:resource="&fibo-der-drc-raw;Warrant"/>
+		<rdfs:label xml:lang="en">currency warrant</rdfs:label>
+		<skos:definition xml:lang="en">warrant that permits the holder to acquire a specified amount in a certain currency at a specified exchange rate during a specified period</skos:definition>
+		<fibo-fnd-utl-av:adaptedFrom xml:lang="en">ISO 10962, Securities and related financial instruments - Classification of financial instruments (CFI) code, Fourth Edition, October 2019</fibo-fnd-utl-av:adaptedFrom>
+	</owl:Class>
+	
+	<owl:Class rdf:about="&fibo-der-drc-raw;DebtWarrant">
+		<rdfs:subClassOf rdf:resource="&fibo-der-drc-raw;Warrant"/>
+		<rdfs:subClassOf rdf:resource="&fibo-der-sbd-sbd;DebtInstrumentDerivative"/>
+		<rdfs:label xml:lang="en">debt warrant</rdfs:label>
+		<skos:definition xml:lang="en">warrant that permits the holder to acquire a specified amount of a debt instrument or related debt observable a specified period at a specified price</skos:definition>
```

Comment:
Not happy about the "or" here - what do the alternatives mean in FIBO terms? No restriction?
word missing before "a specified period".